crates/hir-def/src/resolver.rs

```diff
@@ -245,8 +265,24 @@ impl Resolver {
     pub fn resolve_path_in_value_ns(
         &self,
         db: &dyn DefDatabase,
-        path: &ModPath,
+        path: &Path,
```

Comment:
This likewise isn't too nice, the resolver should not care about generics and what not, so it now taking a `Path` feels a bit weird. (Though maybe its time for us to take another look at how we want to encode hir paths anyways)

Reply:
OTOH it saves a bunch of `.mod_path()` in callers. Maybe it make sense to have another method that gets a `ModPath` and this function call it?

Reply:
> Though maybe its time for us to take another look at how we want to encode hir paths anyways

I would suggest encoding paths as resolved ids and (partially) resolving them in hir lowering. Does it have any problem that current state avoid it?

Reply:
After a second thought this should be fine as well I think

Reply:
I think it simplifies many things. I will try to do it in another PR.